docs/discussions/server-side-graphql.md

```diff
@@ -0,0 +1,123 @@
+
+While Keystone generates the basic CRUD resolvers for you, there are lots of ways you can write custom code which is executed during the resolver phase of a query.
+These include:
+
+- Hooks
+- Access control
+- Virtual field resolver
+- Custom field type resolvers
+- Custom query/mutation resolvers
+
+As a developer you might want to execute some GraphQL operations as part of your custom code.
+This can be done using the function `context.executeGraphQL()`.
+
+The [signature of a resolver function](https://www.apollographql.com/docs/apollo-server/data/resolvers/#resolver-arguments) is `(parent, args, context, info)`.
+The `context` argument is an object which is provided to all resolvers and can contain and state or functions which should be shared across all resolvers.
+Keystone provides a function `executGraphQL()` as part of the `context` which is passed to all resolvers.
+
+This async function takes a `query` and `variables` as arguments and returns `{ data, errors }` as a response.
+
+```
+const { data, errors } = await context.executeGraphQL({ query: '{...}', variables: { ... } });
+```
+
+### Custom context
+
+The function `context.executeGraphQL` will pass `context` through to the resolvers used in the operation.
+There are some instances where you may wish to use a different context to perform the operation.
+For example, you might wish to execute your operation with all access-control turned off, or authenticated as a different user.
+
+This can be achieved using the function `context.createContext({ schemaName, authentication, skipAccessControl })`, which will create a new `context` object based on the supplied arguments.
+
+- `schemaName`: This can be used to select the named schema if you have multiple schemas within your Keystone system.
+- `authentication`: This is an object of the shape `{ item: { id }, listKey: '...' }` which is passed through to access control functions. This can be used to emulate executing the operations as a different user.
+- `skipAccessControl`: This can be used to bypass all access control.
+
+By default each of these values will take the value of the original `context` object.
+You can use this functions like this:
+
+```
+const { data, errors } = await context.executeGraphQL({
+  context: context.createContext({ ... }),
+  query: '{...}',
+  variables: { ... }
+  });
+```
+
+## General purpose operations
+
+There are situations where you might want to execute a GraphQL operations outside of the context of a resolver function.
+These include:
+
+- Migrations
+- Data seeding
+- Responding to a non-GraphQL HTTP request
+- Testing
+- Anything else a developer could possibly imagine
+
+In these cases there is no `context` object for you to borrow, but there is an API to create one for use.
+
+The methods `keystone.createContext()` and `keystone.executeGraphQL()` behave almost exactly the same as the functions `context.createContext()` and `context.executeGraphQL()`. There a few key differences
```

Comment:
I think you meant:
 
```suggestion
The method `keystone.executeQuery()` behaves almost exactly the same as the functions `context.createContext()` and `context.executeGraphQL()` with a few key differences:
```

Reply:
No, I meant what I had originally, I'm comparing each `keystone.foo` to the corresponding `context.foo`.